src/status_im/transport/inbox.cljs

```diff
@@ -219,18 +221,22 @@
  :inbox/request-messages
  (fn [{:keys [db now]} [_ {:keys [from topics discover?]}]]
    (let [web3       (:web3 db)
+         account    (:account/account db)
          wnode      (get-current-wnode-address db)
-         topics     (or topics
-                        (map #(:topic %) (vals (:transport/chats db))))
-         from       (or from (:inbox/last-request db) nil)
-         sym-key-id (:inbox/sym-key-id db)]
-     {::request-messages {:wnode      wnode
-                          :topics     (if discover?
-                                        (conj topics (transport.utils/get-topic constants/contact-discovery))
-                                        topics)
-                          ;;TODO (yenda) fix from, right now mailserver is dropping us
-                          ;;when we send a requestMessage with a from field
-                          ;;:from       from
-                          :sym-key-id sym-key-id
-                          :web3       web3}
-      :db (assoc db :inbox/last-request (quot now 1000))})))
+         topics     (cond-> (or topics
+                                (map #(:topic %) (vals (:transport/chats db))))
+                      discover? (conj (transport.utils/get-topic constants/contact-discovery)))
+         from       (or from (:last-request account))
+         sym-key-id (:inbox/sym-key-id db)
+         now-in-s   (quot now 1000)
+         fx         {:data-store/save-account (assoc account :last-request now-in-s)
+                     :db                      (assoc-in db [:account/account :last-request] now-in-s)}]
+     (if from
+       (when (< delay-in-s-between-mailserver-requests (- now-in-s from))
```

Comment:
As I understand by looking at the code, if we poll the mailserver more than once every 10s we drop the request ( also we reset the counter, which probably should not do). 

Not sure if this is a limitation of the mailserver ( 1 req per 10s), but this might occur in behaviours like:

1) I receive a contact request, poll the mailserver
2) I join a public chat after 5s, no history is fetched

or 

1) I go online, fetch from the mailserver
2) I receive a contact request, no history is fetched

and so on.

It would work if all the requests to the mailserver were identical, but because we can specify a topic, that's not the case.

Why are we limiting the requests, is it something on the mailserver (we get disconnected if more than 1 request per 10 seconds for example), or we are just trying to prevent sending to many requests? (What might be the cause of this, flip-flop of network connection? or too many contact requests/joining public chats?)

Reply:
We don't update the counter when we drop the request because it's done less than 10 sec since the last call to the mailserver it's a when and nothing happens when it's false.

the condition should be more sophisticated though to fetch everytime and not update when there is a topic specify.

This delay avoids spamming the mailserver on flakky connections.

I have thought about other ways to do this over the weekend but there is no easy solution, having one clock per topic wouldn't provide much benefits and would increase complexity/might not even be doable in the future if we send a bloomfilter as a request. 

The case of failing requests is the most problematic as there is no real success for requests, you can't tell if you have received all the messages you requested before losing connectivity for instance.

I think the performance benefits of this PR are on paar with the deduplication PR on go side and only saves more bandwitdth. We can probably delay it to after beta so that we can think more of improvements to request message and consequences of using from parameter

Reply:
Ah you are correct, I got my brackets mixed up! So yes, the only problem would be if mailserver if the connection is down/mailserver is down. But yes, as it is we might lose some messages, so not sure whether we want to include it. What about keeping a single counter and always polling all the topics? That should take care of some edge cases at least, but would also means we don't pull the whole history for a public chat (I remember the test team mentioned we should not fetch history when joining a public chat).